src/components/calculator/util/FormatPrecision.ts

```diff
@@ -1,115 +1,63 @@
-      const digit = Math.floor((val * 10 ** -shift) % 10)
-      if (digit === 9) {
-        return digit
-      }
-      return Math.round((val * 10 ** -shift) % 10)
+    // Returns true if there is a 9 at the 10^n's place.
+    const nineAtPosition = (value: number, shift: number): boolean => {
+      const digit = Math.floor((value * 10 ** -shift) % 10)
+      return digit === 9
     }
 
-    let keepShowingDigits = true
-    let shift = orderOfMagnitude
-    for (; shift >= 0; --shift) {
-      if (keepShowingDigits) {
-        const nextDigit = digitAtPosition(shift)
-        output += keepShowingDigits ? nextDigit : 0
-        keepShowingDigits = nextDigit === 9
+    let shift = 0
+    sigFigsToShow = 1 // Show the first non-9 digit.
+    for (; shift < orderOfMagnitude; ++shift) {
+      // Count the number of nines.
+      if (nineAtPosition(val, orderOfMagnitude - shift)) {
+        sigFigsToShow += 1
       } else {
-        output += '0'
+        break
       }
-      output += commaIfNeeded(shift, thousandsSeparator)
     }
-    if (!keepShowingDigits || !decimalsNearMax) {
-      return output
-    }
-    output += '.'
-    while (true) {
-      const nextDigit = digitAtPosition(shift)
-      output += nextDigit
-      if (nextDigit !== 9) {
-        return output
-      }
-      --shift
+    while (nineAtPosition(val, orderOfMagnitude - shift)) {
+      sigFigsToShow += 1
+      ++shift
     }
   }
-
-  // Round now. Values like 95 will change order of magnitude as a result of rounding.
-  val = Math.round(val * 10 ** -orderOfMagnitude) * 10 ** orderOfMagnitude
-  orderOfMagnitude = Math.floor(Math.log10(val))
-
-  // First digit
-  output += Math.round(val * 10 ** -orderOfMagnitude)
-
-  if (orderOfMagnitude >= 0) {
-    if (orderOfMagnitude % 3 === 0) {
-      output += commaIfNeeded(orderOfMagnitude, thousandsSeparator)
-    }
-    for (let shift = orderOfMagnitude - 1; shift >= 0; --shift) {
-      output += '0'
-      output += commaIfNeeded(shift, thousandsSeparator)
-    }
-    return output
-  }
-
-  // orderOfMagnitude < 0
-  return '0.' + '0'.repeat(-orderOfMagnitude - 1) + output
+  return val.toLocaleString('en-US', {
+    maximumSignificantDigits: sigFigsToShow,
+    style: numberStyle,
+    useGrouping: useGrouping,
+  })
 }
 
 /**
- * Converts |val| to a percentage (including '%') with SIGFIGS sigfigs.
+ * Converts |val| to a percentage (including '%').
  * @param val
  */
-export function fixedPointPrecisionPercent(
-  val: number | null,
-  suffix = '%',
-): string {
-  if (!val) {
-    return '0' + suffix
-  }
-  if (val > 0.9999999) {
-    return '100' + suffix
-  } else if (0.1 < val && val < 0.9) {
-    return (val * 100).toFixed() + suffix
-  }
-  return (
-    fixedPointPrecision(
-      val * 100,
-      /*maxNumber=*/ 100,
-      /*decimalsNearMax=*/ true,
-      /*thousandsSeparator=*/ '',
-    ) + suffix
+export function fixedPointPrecisionPercent(val: number | null): string {
```

Comment:
I don't understand how this works - don't we need to multiply by 100 somewhere?

Reply:
`numberStyle=percent` handles it.